### ISSUE TITLE
Перерисовка грида при движении мышки

### ISSUE THREAD

**Issue by AliakseiDanilchyk** [GoldenSoftwareLtd/gedemin#4521]
Думал, что это уже исправлено, но на exe от 05.12.2024 проблема проявляется.
Подробнее здесь #4500

**Comment by MIchaelShoihet** [comment]
Она стала реже, но все равно случается

**Comment by Polfath** [comment]
Очередная попытка скрыть следы от хинтов грида.   Добавил команду принудительного закрытия хинта. Но все равно нужно проверять.  Мне  баг воспроизвести не удалось.

**Comment by gsbelarus** [comment]
@AliakseiDanilchyk

**Comment by AliakseiDanilchyk** [comment]
Так как нет последовательности действий, чтобы воспроизвести баг, надо какое-то время поэксплуатировать новый exe, чтобы убедиться, что проблема решена.

И еще, может я придираюсь, но тем не менее.
Возможно, это связано с решением проблемы хинтов. Сейчас если мы водим мышкой по колонке грида, в которой не помещается ее содержимое, то весь грид слегка "колбасит". Например, на прилагаемом видео это заметно на заголовке грида.

https://github.com/user-attachments/assets/71da0b5f-6e82-4ae8-8802-4c3313083cca

**Comment by gsbelarus** [comment]
да. это баг с перерисовкой. связанный с тем что туда андрей вставил invalidate для борьбы с хинтами. надо исправлять. не надо все брутально инвалидировать и перерисовывать.

если @Polfath не может справиться пусть @TechAnton посмотрит.

**Comment by Polfath** [comment]
Заменил Invalidate на InvalidateRect. Грид целиком перерисовываться не будет, "колбасить" не должно.

**Comment by AliakseiDanilchyk** [comment]
Так сильно как раньше не колбасит, но подножие грида дергается.
Кроме того, заметил еще один недочет с хинтами.
Если правая граница колонки находится близко к правому краю экрана, и если содержимое колонки слишком длинное, чтобы уместиться в видимой части экрана, то хинт по такой колонке только на долю секунды появляется и сразу исчезает.
На старом exe такая ситуация отрабатывает нормально.

**Comment by YRozmysl** [comment]
Хотел еще заметить, что когда просматриваешь  историю SQL - и перемещаешься по гриду, там, где  запрос был большой и не вмещается на экран, он не отображается.  Не удобно при поиске чего-либо в истории запросов.

![Image](https://github.com/user-attachments/assets/2f2b2f57-252f-4d0a-8119-6bc9f65f0b5f)

Если вмещается запрос - все ок, не вмещается - не ок.

**Comment by TechAnton** [comment]
Новый день и новая попытка исправить ошибку с подсказками. 
Удалил все изменения сделанные ранее, это должно исправить тряску грида и пропадание подсказок, если они не влезают в экран. 
Возможно, вернется проблема с зависшими тенями, описанная ранее (подробнее [здесь](https://github.com/GoldenSoftwareLtd/gedemin/issues/4510) и [здесь](https://github.com/GoldenSoftwareLtd/gedemin/issues/4500)).
Если это произойдет, то напишите пожалуйста, так как есть еще несколько идей как её можно исправить.
Мне её повторить не удается.

**Comment by YRozmysl** [comment]
Вопрос с невлезающими подсказками решился, все остальное надо отслеживать

**Comment by AliakseiDanilchyk** [comment]
Давно уже не повторялся баг. Можно закрывать.
